### PR TITLE
Increase manual throttle resolution by making scaled channels values float

### PR DESCRIPTION
**Pull request by shellixyz**
Complements #18706

### CODE REVIEW

**Comment by amilcarlucas** [comment]
```
Binary           text            data         bss            total
---------------  --------------  -----------  -------------  --------------
arduplane        664 (+0.0384%)  0 (0.0000%)  0 (0.0000%)    664 (+0.0333%)
arducopter       180 (+0.0104%)  0 (0.0000%)  -4 (-0.0015%)  176 (+0.0088%)
ardurover        76 (+0.0049%)   0 (0.0000%)  4 (+0.0015%)   80 (+0.0044%)
antennatracker   268 (+0.0203%)  0 (0.0000%)  4 (+0.0015%)   272 (+0.0172%)
arducopter-heli  212 (+0.0124%)  0 (0.0000%)  4 (+0.0015%)   216 (+0.0110%)
ardusub          44 (+0.0029%)   0 (0.0000%)  -4 (-0.0015%)  40 (+0.0022%)
blimp            184 (+0.0147%)  0 (0.0000%)  0 (0.0000%)    184 (+0.0121%)
```

**Comment by giacomo892** [comment]
Bench tested, all pretty good!
https://ufile.io/1iimjhsa

**Comment by shellixyz** [comment]
I removed the casts for `vfr_hud_throttle()` in Blimp, Copter and Sub

@WickedShell Not sure how these changes would influence throttle DZ if there is any. AFAIK there isn't any in plane anyway

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
I flew this on 3 planes, and Peter flew it on one, no issues found, looks good!

**Comment by rmackay9** [comment]
It looks ok but we need to test on Copter.  For example make sure that the VFR_HUD message looks and also test out compassmot somehow.

**Comment by giacomo892** [comment]
I've flown it on Copter @rmackay9 . Seems like everything is good

**Review by Williangalvani — APPROVED** [review]
This looks fine for Sub

**Comment by magicrub** [comment]
This looks ready to merge so I gave it another look. @WickedShell may have a lingering question about deadzone but I think it's just a statement and not actually a blocker.

My only concern is how it looks on OSD. It was changed to be a float and it will show a bunch of decimal places. It should probably just be cast to an int and not change the display at all. Check draw_throttle()

**Comment by shellixyz** [comment]
It doesn't show decimal places on the OSD because the format is %d

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -1585,7 +1585,7 @@ void AP_OSD_Screen::draw_heading(uint8_t x, uint8_t y)
 
 void AP_OSD_Screen::draw_throttle(uint8_t x, uint8_t y)
 {
-    backend->write(x, y, false, "%3d%c", gcs().get_hud_throttle(), SYMBOL(SYM_PCNT));
+    backend->write(x, y, false, "%3ld%c", lrintf(gcs().get_hud_throttle()), SYMBOL(SYM_PCNT));
```
Comment (magicrub) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1588]: Maybe this is better just casting back to 16int

**Review by magicrub — COMMENTED** [review]
Looks pretty good. I'm wondering if that OSD backend write would be less yeah if just casted back to 16int. Otherwise, looks correct everywhere.

**Pull request closed without merging**